ecommerce/api_test.py

```diff
@@ -201,3 +252,37 @@ def test_unenrollment_unenrolls_learner(mocker, user):
     )
     unenroll_learner_from_order(order_id=order.id)
     unenroll_mock.assert_called()
+
+
+def test_process_cybersource_payment_response(mocker, user_client, user, products):
+    """Test that ensures the response from Cybersource for an ACCEPTed payment updates the orders state"""
+
+    mocker.patch(
+        "mitol.payment_gateway.api.PaymentGateway.validate_processor_response",
+        return_value=True,
+    )
+    create_basket(user, products)
+
+    resp = user_client.post(reverse("checkout_api-start_checkout"))
+
+    payload = resp.json()["payload"]
+    payload = {
+        **{f"req_{key}": value for key, value in payload.items()},
+        "decision": "ACCEPT",
+        "message": "payment processor message",
+    }
+
+    order = Order.objects.get(state=Order.STATE.PENDING, purchaser=user)
+
+    assert order.reference_number == payload["req_reference_number"]
+
+    request = HttpRequest()
+    request.method = "POST"
+    request.POST = payload
```

Comment:
Django has a [`RequestFactory`](https://docs.djangoproject.com/en/3.2/topics/testing/advanced/#django.test.RequestFactory) helper and `pytest` [wraps it in a convenient fixture](https://pytest-django.readthedocs.io/en/latest/helpers.html#rf-requestfactory), so you can simplify this down to `rf.post("/url", payload)`